### PR TITLE
Slider update

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
# What does this PR do? Why are doing this? References?
range_slider module is updated upon review. Getter and setter of values changed to be compatible with Qt style. Demo part is separated from module and given as an example to use the slider.

## Type of change
[Please delete options that are not relevant.]
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Only manually tested on OSX Mojave with python 3.7.0 and PyQt5 5.10.1.
- [X] manual test - examples/slider.py

## Final Checklist:
- [X] My PR is the possible minimum work for the desired functionality
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: examples/slider.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import sys
+from PyQt5 import QtWidgets
+
+from ..gui.util.range_slider import QHRangeSlider
```
Comment (kne42) [inline examples/slider.py]: this shouldn't be a relative import

**Review by kne42 — CHANGES_REQUESTED** [review]

File: examples/slider.py (hunk 1 above)
Comment (jni) [reply examples/slider.py]: @kne42 why not?

Comment (jni) [reply examples/slider.py]: Oh never mind, just noticed why. =P

File: gui/util/range_slider.py [hunk 2]
```diff
@@ -42,9 +42,9 @@ def __init__(self, slider_range, values, parent=None):
         else:
             self.setRange([0.0, 1.0, 0.01])
         if values:
-            self.values = values
+            self.setValues(values)
         else:
-            self.values = [0.3, 0.6]
+            self.setValues([0.3, 0.6])
```
Comment (jni) [inline gui/util/range_slider.py:47]: @kne42 can you elaborate on why this change is needed?

**Comment by jni** [comment]
@AhmetCanSolak I don't really agree with the choice of using a dev branch. I realise the old "git flow" post advocates for one, but most (all?) packages in the scientific Python ecosystem seem to get by just fine with just a working master. Do you think it is necessary here? My intuition is that it will simply drive more merge conflicts.

File: gui/util/range_slider.py (hunk 2 above)
Comment (kne42) [reply gui/util/range_slider.py:47]: @jni This was using properties before, which is a Pythonic implementation. To create a "truly" Qt-style class, we should be using get and set functions.

**Comment by kne42** [comment]
@jni We decided on this approach because we expect there to be rapid, iterative development on the dev branch with potentially breaking changes from commit to commit. We want to separate this from the master branch with "early" releases.

This allows the master branch to be relatively stable from change to change, and grants users the ability to effectively subscribe to the upcoming release without necessarily breaking their installation.

While conflicts may be a problem, we expect dev branch contributors to be pretty experienced with both git and Python and easily able to resolve and/or adapt to them. All changes to master should be squash-merged regardless to keep the package easily git-installable, which will cover up a lot of the messiness that merge commits bring.

File: gui/util/range_slider.py (hunk 2 above)
Comment (jni) [reply gui/util/range_slider.py:47]: I'm vaguely ok with a Pythonic "falsely" Qt class, as long as it works... ;) But I am not going to die on this particular hill. =P

**Comment by jni** [comment]
@kne42 if by "breaking" changes you mean that the API changes, I think this is totally fine on master, and doesn't require a new branch. This is simply the stage of development and no one should expect a master API to be stable.

If, on the other hand, you mean that stuff on the dev branch may not even be functional... Well, I don't see much use in that either, since that sort of work should happen on feature branches anyway.

Having said all this: whatever works for you guys now, since you are much more active in development than me. I **hope** that this will change very soon though, at which point I might be quite a bit more opinionated about the branching model. =P

**Comment by royerloic** [comment]
I agree with Kira, we had a few examples of people wanting to try Napari only to be temporarily discouraged by a broken master. I would suggest we keep master for releases and develop on master. 
I like the idea of topic branches, which can be a good alternative to a dev branch...

**Review by royerloic — APPROVED** [review]

**Comment by AhmetCanSolak** [comment]
@jni thank you for your comment.
I think the traditional git flow is `perfect` from many perspectives and we have some cases as @royerloic explained for motivation to keep branches separated. I know majority(almost all) of the python packages does not follow such good practices, especially the ones in scientific community :) it is worth to question, I believe.

We are also planning to use master as a release branch which I believe will be so helpful for the git-based plugins of Napari.

**Comment by jni** [comment]
@AhmetCanSolak sure, happy for it to be questioned, but I'll point out that the SciPy "master-only" model came *after* Fernando Perez and others questioned whether git-flow was necessary, or just overhead. Many of these projects started out using git-flow then got to the point that dev could nicely act as master anyway. Specifically, if we have good test coverage and process, master should never be broken. dev to me seems to encourage breaking things.

I also don't like the sound of squashing everything in between releases. A clean history is absolutely critical for finding regressions and when things broke.

**Review by kne42 — APPROVED** [review]
